index.js

```diff
@@ -129,11 +127,28 @@ module.exports.init = function () {
       unix_sockets.push(address.address);
       fs.chmodSync(address.address, '664');
     }
+  }).on('exit', function (worker, code, signal) {
+    if (worker.exitedAfterDisconnect) { // since: Node v6.0.0
```

Comment:
When master-process kills a worker it will now use [`Worker.kill`](http://nodejs.org/api/cluster.html#cluster_worker_kill_signal_sigterm) instead of [`process.kill`](http://nodejs.org/api/process.html#process_process_kill_pid_signal). This sets `Worker.exitedAfterDisconnect`, which we use here to distinguish unexpected exits from expected ones.